### ISSUE TITLE
cmd/compile/internal/syntax: TestStdLib verbosely broken on Windows

### ISSUE THREAD

**Issue by bcmills** [golang/go#35175]
https://build.golang.org/log/1d9e9c5f8db18b185d84dffca1c0372654dfcfa1 shows an amazingly long test failure on the `windows-amd64-longtest` builder.

I can't make heads or tails of it myself, and I don't know how long it's been broken, but it seems clear that something is going very wrong in the test on Windows.

CC @alexbrainman @zx2c4

**Comment by bcmills** [comment]
Ah, here we go:
```
--- FAIL: TestStdLib (20.36s)
    parser_test.go:54: C:\workdir\go\test\bombad.go:14:2: invalid BOM in the middle of the file
    parser_test.go:54: C:\workdir\go\test\char_lit1.go:16:13: escape sequence is invalid Unicode code point U+D800
    parser_test.go:54: C:\workdir\go\test\func3.go:16:8: syntax error: mixed named and unnamed function parameters
    parser_test.go:54: C:\workdir\go\test\slice3err.go:20:14: middle index required in 3-index slice
    parser_test.go:54: C:\workdir\go\test\switch2.go:14:8: syntax error: unexpected semicolon, expecting :
    parser_test.go:54: C:\workdir\go\test\fixedbugs\bug014.go:10:21: non-octal character in escape sequence: '
    parser_test.go:54: C:\workdir\go\test\fixedbugs\bug050.go:7:1: syntax error: package statement must be first
    parser_test.go:54: C:\workdir\go\test\fixedbugs\bug068.go:12:13: unknown escape sequence
    parser_test.go:54: C:\workdir\go\test\fixedbugs\bug088.go:7:1: syntax error: package statement must be first
    parser_test.go:54: C:\workdir\go\test\fixedbugs\bug106.go:7:1: syntax error: package statement must be first
    parser_test.go:54: C:\workdir\go\test\fixedbugs\bug121.go:12:7: syntax error: name list not allowed in interface type
    parser_test.go:54: C:\workdir\go\test\fixedbugs\bug163.go:10:3: invalid identifier character U+229B '⊛'
    parser_test.go:54: C:\workdir\go\test\fixedbugs\bug169.go:8:10: empty character literal or unescaped ' in character literal
    parser_test.go:54: C:\workdir\go\test\fixedbugs\bug217.go:12:10: method has no receiver
    parser_test.go:54: C:\workdir\go\test\fixedbugs\bug222.go:7:1: syntax error: package statement must be first
    parser_test.go:54: C:\workdir\go\test\fixedbugs\bug228.go:11:30: syntax error: final argument in variadic function missing type
    parser_test.go:54: C:\workdir\go\test\fixedbugs\bug274.go:23:5: syntax error: missing statement after label
    parser_test.go:54: C:\workdir\go\test\fixedbugs\bug282.go:7:1: syntax error: package statement must be first
    parser_test.go:54: C:\workdir\go\test\fixedbugs\bug298.go:10:2: syntax error: non-declaration statement outside function body
    parser_test.go:54: C:\workdir\go\test\fixedbugs\bug299.go:16:64: syntax error: cannot parenthesize embedded type
    parser_test.go:54: C:\workdir\go\test\fixedbugs\bug300.go:23:16: syntax error: cannot parenthesize type in composite literal
    parser_test.go:54: C:\workdir\go\test\fixedbugs\bug306.go:7:1: syntax error: package statement must be first
    parser_test.go:54: C:\workdir\go\test\fixedbugs\bug349.go:12:14: syntax error: unexpected literal 2.01 at end of statement
    parser_test.go:54: C:\workdir\go\test\fixedbugs\bug388.go:12:9: syntax error: mixed named and unnamed function parameters
    parser_test.go:54: C:\workdir\go\test\fixedbugs\bug394.go:10:1: syntax error: non-declaration statement outside function body
    parser_test.go:54: C:\workdir\go\test\fixedbugs\bug435.go:15:48: syntax error: unexpected EOF, expecting )
    parser_test.go:54: C:\workdir\go\test\fixedbugs\issue11359.go:11:5: identifier cannot begin with digit U+06F6 '۶'
    parser_test.go:54: C:\workdir\go\test\fixedbugs\issue11610.go:12:4: invalid character U+003F '?'
    parser_test.go:54: C:\workdir\go\test\fixedbugs\issue13248.go:13:1: syntax error: unexpected }, expecting expression
    parser_test.go:54: C:\workdir\go\test\fixedbugs\issue13266.go:10:8: syntax error: unexpected %, expecting name
    parser_test.go:54: C:\workdir\go\test\fixedbugs\issue13273.go:50:11: syntax error: unexpected <-, expecting chan
    parser_test.go:54: C:\workdir\go\test\fixedbugs\issue13274.go:11:43: syntax error: unexpected EOF, expecting }
    parser_test.go:54: C:\workdir\go\test\fixedbugs\issue13319.go:12:65: syntax error: unexpected newline, expecting comma or )
    parser_test.go:54: C:\workdir\go\test\fixedbugs\issue14006.go:24:4: syntax error: unexpected : at end of statement
    parser_test.go:54: C:\workdir\go\test\fixedbugs\issue14164.go:7:1: syntax error: package statement must be first
    parser_test.go:54: C:\workdir\go\test\fixedbugs\issue14520.go:12:14: syntax error: unexpected newline, expecting comma or )
    parser_test.go:54: C:\workdir\go\test\fixedbugs\issue15611.go:15:7: empty character literal or unescaped ' in character literal
    parser_test.go:54: C:\workdir\go\test\fixedbugs\issue17328.go:11:13: syntax error: unexpected ), expecting { after for clause
    parser_test.go:54: C:\workdir\go\test\fixedbugs\issue18092.go:14:36: syntax error: unexpected newline, expecting :
    parser_test.go:54: issue18393.go:24[C:\workdir\go\test\fixedbugs\issue18393.go:17:38]: syntax error: unexpected newline, expecting type
    parser_test.go:54: C:\workdir\go\test\fixedbugs\issue18747.go:10:5: syntax error: missing condition in if statement
    parser_test.go:54: C:\workdir\go\test\fixedbugs\issue18915.go:13:13: syntax error: a := 10 used as value
    parser_test.go:54: C:\workdir\go\test\fixedbugs\issue19056.go:9:9: syntax error: unexpected ..., expecting expression
    parser_test.go:54: C:\workdir\go\test\fixedbugs\issue19610.go:10:12: syntax error: cannot declare in post statement of for loop
    parser_test.go:54: C:\workdir\go\test\fixedbugs\issue19667.go:13:1: syntax error: unexpected }, expecting expression
    parser_test.go:54: C:\workdir\go\test\fixedbugs\issue20789.go:13:21: syntax error: unexpected u, expecting (
    parser_test.go:54: C:\workdir\go\test\fixedbugs\issue22164.go:12:42: syntax error: unexpected newline, expecting comma or )
    parser_test.go:54: C:\workdir\go\test\fixedbugs\issue22581.go:10:13: syntax error: unexpected ), expecting {
    parser_test.go:54: C:\workdir\go\test\fixedbugs\issue23586.go:21:8: expression in defer must be function call
    parser_test.go:54: C:\workdir\go\test\fixedbugs\issue23587.go:10:6: invalid character U+007E '~'
    parser_test.go:54: C:\workdir\go\test\fixedbugs\issue23664.go:12:9: syntax error: unexpected true, expecting {
    parser_test.go:54: C:\workdir\go\test\fixedbugs\issue24801.go:6:1: syntax error: package statement must be first
    parser_test.go:54: C:\workdir\go\test\fixedbugs\issue30722.go:12:7: '_' must separate successive digits
    parser_test.go:54: C:\workdir\go\test\fixedbugs\issue32133.go:10:12[C:\workdir\go\test\fixedbugs\issue32133.go:16:12]: newline in string
    parser_test.go:54: C:\workdir\go\test\fixedbugs\issue33386.go:16:2: syntax error: unexpected }, expecting expression
    parser_test.go:54: C:\workdir\go\test\fixedbugs\issue4405.go:11:3: invalid character U+0007
    parser_test.go:54: C:\workdir\go\test\fixedbugs\issue4468.go:22:5: expression in go must be function call
    parser_test.go:54: C:\workdir\go\test\fixedbugs\issue4776.go:9:1: syntax error: package statement must be first
    parser_test.go:54: C:\workdir\go\test\fixedbugs\issue9036.go:21:13: 'p' exponent requires hexadecimal mantissa
    parser_test.go:54: C:\workdir\go\test\syntax\chan.go:11:1: syntax error: missing channel element type
    parser_test.go:54: C:\workdir\go\test\syntax\chan1.go:13:12: syntax error: c <- v used as value
    parser_test.go:54: C:\workdir\go\test\syntax\composite.go:10:91: syntax error: unexpected newline, expecting comma or }
    parser_test.go:54: C:\workdir\go\test\syntax\ddd.go:10:6: syntax error: unexpected literal .3, expecting name or (
    parser_test.go:54: C:\workdir\go\test\syntax\else.go:11:9: syntax error: else must be followed by if or statement block
    parser_test.go:54: C:\workdir\go\test\syntax\if.go:13:5: syntax error: missing condition in if statement
    parser_test.go:54: C:\workdir\go\test\syntax\import.go:10:6: syntax error: unexpected comma, expecting semicolon or newline or )
    parser_test.go:54: C:\workdir\go\test\syntax\initvar.go:10:9: syntax error: var declaration not allowed in if initializer
    parser_test.go:54: C:\workdir\go\test\syntax\interface.go:10:7: syntax error: name list not allowed in interface type
    parser_test.go:54: C:\workdir\go\test\syntax\semi1.go:10:62: syntax error: unexpected newline, expecting { after if clause
    parser_test.go:54: C:\workdir\go\test\syntax\semi2.go:10:68: syntax error: missing { after switch clause
    parser_test.go:54: C:\workdir\go\test\syntax\semi3.go:10:66: syntax error: unexpected newline, expecting { after for clause
    parser_test.go:54: C:\workdir\go\test\syntax\semi4.go:11:2: syntax error: unexpected {, expecting for loop condition
    parser_test.go:54: C:\workdir\go\test\syntax\semi5.go:10:1: syntax error: unexpected semicolon or newline before {
    parser_test.go:54: C:\workdir\go\test\syntax\semi6.go:9:58: syntax error: unexpected newline in type declaration
    parser_test.go:54: C:\workdir\go\test\syntax\semi7.go:11:2: syntax error: unexpected else, expecting }
    parser_test.go:54: C:\workdir\go\test\syntax\topexpr.go:9:1: syntax error: non-declaration statement outside function body
    parser_test.go:54: C:\workdir\go\test\syntax\typesw.go:10:43: syntax error: main() := interface{}(nil).(type) used as value
    parser_test.go:54: C:\workdir\go\test\syntax\vareq.go:10:25: syntax error: unexpected { at end of statement
    parser_test.go:54: C:\workdir\go\test\syntax\vareq1.go:9:24: syntax error: unexpected { after top level declaration
package syntax
```

**Comment by griesemer** [comment]
The actual error was a silly oversight for Windows (hardwired path separator). The verbose long test output is independent. Will fix separately (see #35223).

**Comment by dmitshur** [comment]
@gopherbot, please consider this issue for backport to Go 1.13. This is a test fix. It is needed to resolve a test failure in the `cmd/compile/internal/syntax` package on windows-amd64-longtest builder, which could be masking other problems (due to #36181) and making releasing Go 1.13.x more difficult.
